python/delta_sharing/rest_client.py

```diff
@@ -180,3 +190,30 @@ def _request_internal(self, request, target: str, data: Optional[Dict[str, Any]]
                 collections.deque(lines, maxlen=0)
         finally:
             response.close()
+
+    def _retry_with_exponential_backoff(self, func):
+        times_retried = 0
+        sleep_ms = 100
+        while True:
+            times_retried += 1
+            try:
+                return func()
+            except Exception as e:
```

Comment:
I guess we can use `HTTPError` here if we re-throw the exception when it's not `HTTPError`?

Reply:
Then we can focus on only `HTTPError` in `_should_retry`.

Reply:
The scala version also retries when an IOException is hit. I wasn't sure if we want to do the same for Python, so I asked in my comment above https://github.com/delta-io/delta-sharing/pull/49#discussion_r673571543